### PR TITLE
Tweak OSX XInput layout

### PR DESCRIPTION
**Pull request by douglas-piconi**
The normalization values were screwing up the sticks, and the Up/Down had the wrong layout.

### CODE REVIEW

**Review by Rene-Damm — CHANGES_REQUESTED** [review]
Need test.

**Comment by douglas-piconi** [comment]
@Rene-Damm tests added.

File: Packages/com.unity.inputsystem/Tests/InputSystem/Plugins/XInputTests.cs [hunk 1]
```diff
@@ -127,5 +130,132 @@ public void Devices_SupportsControllerOnXbox()
...
+            new MICEXboxControllerForOSXState
+            {
+                leftX = -32767,
+                leftY = -32767,
+            });
+
+        InputSystem.Update();
+        Assert.That(gamepad.leftStick.x.ReadValue(), Is.EqualTo(-0.9999).Within(0.001));
+        Assert.That(gamepad.leftStick.y.ReadValue(), Is.EqualTo(0.9999).Within(0.001));
+    }
```
Comment (Rene-Damm) [inline Packages/com.unity.inputsystem/Tests/InputSystem/Plugins/XInputTests.cs:223]: Add right stick and trigger and use AssetButtonPress to check buttons.

File: Packages/com.unity.inputsystem/Tests/InputSystem/Plugins/XInputTests.cs [hunk 2]
```diff
@@ -127,5 +130,132 @@ public void Devices_SupportsControllerOnXbox()
...
+            new XINPUT_GAMEPADState
+            {
+                leftX = -32767,
+                leftY = 32767,
+            });
+
+        InputSystem.Update();
+        Assert.That(gamepad.leftStick.x.ReadValue(), Is.EqualTo(-0.9999).Within(0.001));
+        Assert.That(gamepad.leftStick.y.ReadValue(), Is.EqualTo(0.9999).Within(0.001));
+    }
```
Comment (Rene-Damm) [inline Packages/com.unity.inputsystem/Tests/InputSystem/Plugins/XInputTests.cs:316]: Same here as with the test above.

**Review by Rene-Damm — CHANGES_REQUESTED** [review]

File: Packages/com.unity.inputsystem/Tests/InputSystem/Plugins/XInputTests.cs [hunk 3]
```diff
@@ -127,5 +130,190 @@ public void Devices_SupportsControllerOnXbox()
...
+        [FieldOffset(6)] public short leftX;
+        [FieldOffset(8)] public short leftY;
+        [FieldOffset(10)] public short rightX;
+        [FieldOffset(12)] public short rightY;
+
+        public FourCC GetFormat()
+        {
+            return new FourCC('H', 'I', 'D');
+        }
+    }
```
Comment (Rene-Damm) [inline Packages/com.unity.inputsystem/Tests/InputSystem/Plugins/XInputTests.cs:156]: Let's move the structs into the XInput plugin next to replace the current JSON layouts.

**Pull request merged**